templates/commands/render/action.go

```diff
@@ -141,6 +141,29 @@ func templateAndCompileRegexes(regexes []model.String, scope *common.Scope) ([]*
 	return compiled, merr
 }
 
+// processPaths processes a list of input String paths for go templating, relative paths,
+// OS-specific slashes, and (soon) file globbing.
+func processPaths(paths []model.String, scope *common.Scope) ([]model.String, error) {
```

Comment:
Can we add a test that ensures the given arguments are not modified? Basically make a copy of `paths`, call the function, ensure paths has not changed.

Reply:
Added test for this + some extras for processPaths.